Source/p_pspr.c

```diff
@@ -1065,7 +1065,25 @@ void P_MovePsprites(player_t *player)
   psp = &player->psprites[ps_weapon];
   psp->sx2 = psp->sx;
   psp->sy2 = psp->sy;
-  if (psp->state && center_weapon)
+
+  if (psp->state && !cosmetic_bobbing)
+  {
+    static fixed_t last_sy = 32 * FRACUNIT;
+
+    psp->sx2 = FRACUNIT;
+
+    if (psp->state->action != A_Lower && psp->state->action != A_Raise)
+    {
+      last_sy = psp->sy2;
+      psp->sy2 = 32 * FRACUNIT;
+    }
+    else if (psp->state->action == A_Lower)
+    {
+      // We want to move smoothly from where we were
+      psp->sy2 -= (last_sy - 32 * FRACUNIT);
+    }
+  }
+  else if (psp->state && center_weapon)
```

Comment:
If cosmetic bobbing and centered weapon sprite exclude each other, we should make this more obvious in the menu.

Reply:
Fixed in https://github.com/fabiangreffrath/woof/pull/528/commits/b1fef3675169126e2acfabc84f76bca0d4eb57ba